File: dataset/human/java/file_1052.java
import java.util.*;

public class Main {

	static final double EPS = 1e-8;

	class Point{
		double x, y;
		Point(double _x, double _y){
			this.x = _x;
			this.y = _y;
		}
		Point add(Point a){
			return new Point(this.x + a.x, this.y + a.y);
		}
		Point sub(Point a){
			return new Point(this.x - a.x, this.y - a.y);
		}
		double abs(){
			return Math.hypot(this.x, this.y);
		}
	}

	double iprod(Point p, Point q){
		return p.x*q.x + p.y*q.y;
	}

	double oprod(Point p, Point q){
		return p.x*q.y - p.y*q.x;
	}

	double distanceLinePoint(Point p1, Point p2, Point q){
		return Math.abs(oprod(p2.sub(p1), q.sub(p1))) / p2.sub(p1).abs();
	}

	double distanceSegmentPoint(Point p1, Point p2, Point q){
		if(iprod(p2.sub(p1), q.sub(p1)) < EPS) return q.sub(p1).abs();
		if(iprod(p1.sub(p2), q.sub(p2)) < EPS) return q.sub(p2).abs();
		return distanceLinePoint(p1,p2,q);
	}

	Point[] Cs;
	double[] Rs;

	void run(){
		Scanner in = new Scanner(System.in);
		for(int n=in.nextInt(); n>0; n = in.nextInt()){
			Cs = new Point[n];
			Rs = new double[n];
			for(int i=0; i<n; i++){
				double x = in.nextDouble(), y = in.nextDouble(), r = in.nextDouble();
				Cs[i] = new Point(x,y);
				Rs[i] = r;
			}
			int m = in.nextInt();
			for(int i=0; i<m; i++){
				double tx = in.nextDouble(), ty = in.nextDouble(),
						sx = in.nextDouble(), sy = in.nextDouble();
				System.out.println(solve(new Point(tx, ty), new Point(sx, sy))?"Danger":"Safe");
			}
		}
	}

	boolean check(Point p, Point q, Point center, double r){
		if((p.sub(center).abs() < r - EPS)^(q.sub(center).abs() < r - EPS)){
			return false;
		}
		if(p.sub(center).abs() < r - EPS){
			return true;
		}
		return distanceSegmentPoint(p,q,center) > r + EPS;
	}

	boolean solve(Point p, Point q){
		for(int i=0; i<Cs.length; i++){
			if(!check(p, q, Cs[i], Rs[i])) return false;
		}
		return true;
	}

	public static void main(String args[]){
		new Main().run();
	}
}

File: dataset/human/python/file_1052.py
import math as M
def R(A): return (A[0]**2+A[1]**2)**.5
def I(i): return [map(int,raw_input().split()) for _ in [0]*i]
def C(a,b): return a>b or abs(a-b)<1e-6
def f(e1):
  tx,ty,sx,sy=e1
  x=[]
  st=[tx-sx,ty-sy]; rst=R(st)
  for e2 in WP:
    wx,wy,r=e2
    wt=[tx-wx,ty-wy]; rwt=R(wt)
    sw=[wx-sx,wy-sy]; rsw=R(sw)
    F=[rwt<r,rsw<r]
    if F==[1,0] or F==[0,1]: return 0
    elif F==[0,0]:
      a=M.pi/2-M.acos(r/rsw)
      b=M.acos(round((sw[0]*st[0]+sw[1]*st[1])/rsw/rst,4))
      if C(a,b) and C(rst**2,rsw**2-r**2): return 0
  return 1

while 1:
  n=input()
  if n==0: break
  WP=I(n)
  P=I(input())
  for e in P: print ["Safe","Danger"][f(e)]
